test/transactions/utils/signAndVerify.js

```diff
@@ -250,3 +250,1410 @@ describe('signAndVerify transaction utils', () => {
+					signSignature:
+						'ac3a9eec518b8c725aa8f8620546001f72b3036078242ab0c7639b6d6891ba4330448cc3e4f5c31516e896af4c5fa998fc2ba2673655a4cfb76168c8785c9109',
+					id: '2884334432991037421',
+				},
+				{
+					type: 5,
+					amount: '0',
+					fee: '2500000000',
+					recipientId: null,
+					senderPublicKey:
+						'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+					timestamp: 54196080,
+					asset: {
+						dapp: {
+							category: 0,
+							name: 'OpwYfwHdAfO4ncn7D',
+							description: 'Ds',
+							tags: '1nkfYVcgsisKnXyn7k0',
+							type: 0,
+							link: 'WQDusb0DgH',
+							icon: 'RPAFQsBIsE',
+						},
+					},
+					signature:
+						'524afb27d284e4e71ea44de9d23f9a1cd603f37f81a55187a61ca92391dce1994d2c4a5e3f0ae8490caac66da5125a0d03f30d0775592aa02d451a72e3ed9303',
+					signSignature:
+						'7043b795dab467e3d3239e7c379ee1b07914a8ba04e639bda406f6ed8810d75a7b4066ad5e90ef3c2030927b917e8492db59ffad017ce51878b7217b27d3a506',
+					id: '7976119586785833934',
+				},
+				{
+					type: 6,
+					amount: '881374',
+					fee: '10000000',
+					recipientId: null,
+					senderPublicKey:
+						'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+					timestamp: 54196080,
+					asset: { inTransfer: { dappId: '1000000' } },
+					signature:
+						'f9facd4dd0b278e3fdca44e6b5527ff4d71f65fb05d1a993cc094f7a6e08118d96ad9103c7577c9efff41d3deec08ffb06346ab27aeff800d499785b53bea40d',
+					signSignature:
+						'ddbe2671cb80cf478b581772242e20d49b8f88445928f8c685ea06a83733a7c7abec31c53f52f1962126abc6e1e03b2c8c7b881f6a3d6808496eb500498b730b',
+					id: '10748675444711410396',
+				},
+				{
+					type: 7,
+					amount: '370999',
+					fee: '10000000',
+					recipientId: '1859190791819301L',
+					senderPublicKey:
+						'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+					timestamp: 54196080,
+					asset: {
+						outTransfer: { dappId: '763580659', transactionId: '89998498' },
+					},
+					signature:
+						'8041323c405e23fc8ebaaf7ded1a61e1922c29643d6ab5cf536655fb69380d7e64a158198a21cd34faa06ed1777321f403421fe59fc1b4c40087c19ab47eda0c',
+					signSignature:
+						'177664c1e4afc88462f44f8af6192d36d435bf9457d2466e9ced8b52d44eb740be03232a6a9e0467762cf336bbd155fba04dd0a74cb7a6938ee976453bf6070b',
+				},
+			];
+		});
+
+		describe('#signTransaction', () => {
+			describe('given a passphrase and a second passphrase', () => {
+				const passphrase =
+					'wagon stock borrow episode laundry kitten salute link globe zero feed marble';
+				const secondPassphrase =
+					'trouble float modify long valve group ozone possible remove dirt bicycle riot';
+
+				describe('when tested on the first signature', () => {
+					it('should create the same signature', done => {
+						transactions.forEach(transaction => {
+							const { signature } = transaction;
+							const rawTx = Object.assign({}, transaction);
```

Comment:
This is fine for signatures, but we should be careful if we extend this test pattern to assets because the nested objects will have the same reference.